src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -53,10 +53,10 @@ public function process(ContainerBuilder $container)
             $container->removeDefinition('messenger.middleware.debug.logging');
         }
 
-        if (!$container->has('serializer')) {
-            $container->removeDefinition('messenger.transport.serialize_message_with_type_in_headers');
-            $container->removeAlias('messenger.transport.default_encoder');
-            $container->removeAlias('messenger.transport.default_decoder');
+        if (!$container->has('serializer') && $container->hasAlias('messenger.transport.encoder')) {
+            if ('messenger.transport.symfony_serializer_encoder' === (string) $container->getAlias('messenger.transport.encoder')) {
+                throw new RuntimeException('Using the default encoder/decoder, Symfony Messenger requires the Serializer. Try running "composer require symfony/serializer-pack".');
```

Comment:
`\LogicException`

Reply:
That could also happen when the serializer component is simply disabled (but sure, using flex the component is enabled by default when installed).

However, what about throwing directly from the `FrameworkExtension`? When `$this->isConfigEnabled($container, $config['transport']['serializer'])` is `true` and the `$this->isConfigEnabled($container, $config['serializer'])` is `false`, we should throw an exception about enabling the component. If trying to enabling it without the component installed, the Fwb ext will already hint for installing the component.

The `'messenger.transport.symfony_serializer_encoder' === (string) $container->getAlias('messenger.transport.encoder')` check doesn't look really relevant to me. If the `messenger.transport.serializer` config is enabled without the Serializer component installed/enabled, it should throw anyway.

Reply:
> However, what about throwing directly from the FrameworkExtension? When $this->isConfigEnabled($container, $config['transport']['serializer']) is true and the $this->isConfigEnabled($container, $config['serializer']) is false, we should throw an exception about enabling the component. If trying to enabling it without the component installed, the Fwb ext will already hint for installing the component.

Very good idea!